File: 0804-unique-morse-code-words/0804-unique-morse-code-words.rs
use std::collections::{HashSet};

const MORSE_CODE: [&str; 26]  = [".-","-...","-.-.","-..",".","..-.","--.","....","..",".---","-.-",".-..","--","-.","---",".--.","--.-",".-.","...","-","..-","...-",".--","-..-","-.--","--.."];
const BASE_CHAR: u8 = 'a'.to_ascii_lowercase() as u8;

impl Solution {
    pub fn unique_morse_representations(words: Vec<String>) -> i32 {
        let mut set = HashSet::<String>::new();        

        for word in words.iter() {
            let mut str = String::new();

            for c in word.chars().into_iter() {
                let ascii = c.to_ascii_lowercase() as u8 - BASE_CHAR;
                let mc = MORSE_CODE.get(ascii as usize).unwrap();

                str.push_str(mc);
            }

            set.insert(str);
        }

        set.len() as i32
    }
}

File: 0804-unique-morse-code-words/0804-unique-morse-code-words.swift
let MORSE_CODE: [String] = [".-","-...","-.-.","-..",".","..-.","--.","....","..",".---","-.-",".-..","--","-.","---",".--.","--.-",".-.","...","-","..-","...-",".--","-..-","-.--","--.."]

class Solution {
    func uniqueMorseRepresentations(_ words: [String]) -> Int {
        var unique_code: Set<String> = []
        
        let base: Character = "a"
        for word in words {
            var morseCode = ""
            
            for c in word {
                let index = c.asciiValue! - base.asciiValue!
                let mc = MORSE_CODE[Int(index)]
                morseCode.append(mc)
            }
            
            unique_code.insert(morseCode)
        }
        
        return unique_code.count
    }
}

